diff --git a/projects/MTK/src/stk_impl/cl_MTK_Mesh_Core_STK.cpp b/projects/MTK/src/stk_impl/cl_MTK_Mesh_Core_STK.cpp
index b0c419f..1a26e6b 100644
--- a/projects/MTK/src/stk_impl/cl_MTK_Mesh_Core_STK.cpp
+++ b/projects/MTK/src/stk_impl/cl_MTK_Mesh_Core_STK.cpp
@@ -14,6 +14,7 @@
 #include <stk_mesh/base/GetEntities.hpp>          // for count_entities
 #include <stk_mesh/base/Selector.hpp>             // for Selector
 #include <stk_mesh/base/FEMHelpers.hpp>           // for Selector
+#include <stk_mesh/base/MeshBuilder.hpp>          // for MeshBuilder
 #include "stk_io/DatabasePurpose.hpp"             // for DatabasePurpose::READ_MESH
 #include "stk_mesh/base/CoordinateSystems.hpp"    // for Cartesian
 #include "stk_mesh/base/CreateFaces.hpp"          // for handling faces
@@ -91,10 +92,12 @@ namespace moris
             MPI_Comm aCommunicator = MPI_COMM_WORLD;
 
             // Generate MetaData and Bulk Data instances (later to be pointed to member variables)
-            stk::mesh::MetaData* meshMeta = new stk::mesh::MetaData;
+            auto meshMeta = std::make_shared<stk::mesh::MetaData>();
 
             MORIS_LOG( "Construct Bulk Data" );
-            stk::mesh::BulkData* meshBulk = new stk::mesh::BulkData( *meshMeta, aCommunicator, this->get_aura_option() );
+            auto meshBuilder = stk::mesh::MeshBuilder{ aCommunicator };
+            meshBuilder.set_aura_option( this->get_aura_option() );
+            stk::mesh::BulkData* meshBulk = meshBuilder.create( meshMeta ).release();
 
             // Set member variables as pointers to meta_data and bulk_data
             mSTKMeshData->mMtkMeshMetaData = ( meshMeta );
@@ -131,7 +134,7 @@ namespace moris
             mSTKMeshData->mMeshReader->populate_bulk_data();
 
             // Determine number of time increments on input database in region
-            Teuchos::RCP< Ioss::Region > tIo_region      = mSTKMeshData->mMeshReader->get_input_io_region();
+            std::shared_ptr< Ioss::Region > tIo_region   = mSTKMeshData->mMeshReader->get_input_ioss_region();
             int                          tTimestep_count = tIo_region->get_property( "state_count" ).get_int();
 
             MORIS_LOG( "Read Input Fields" );
@@ -2577,7 +2580,7 @@ namespace moris
             //  can be declared between two parts, and a field definition can be limited to specific parts.
 
             // Declare and initialize Stk mesh
-            stk::mesh::MetaData* meshMeta = new stk::mesh::MetaData( mSTKMeshData->mNumDims );
+            auto meshMeta = std::make_shared< stk::mesh::MetaData >( mSTKMeshData->mNumDims );
 
             // Set member variable as pointer to meta_data
             mSTKMeshData->mMtkMeshMetaData = meshMeta;
@@ -2606,7 +2609,9 @@ namespace moris
             stk::ParallelMachine tPM = MPI_COMM_WORLD;
 
             // Create BulkData Object
-            stk::mesh::BulkData* meshBulk = new stk::mesh::BulkData( *mSTKMeshData->mMtkMeshMetaData, tPM, this->get_aura_option() );
+            auto meshBuilder = stk::mesh::MeshBuilder{ tPM };
+            meshBuilder.set_aura_option( this->get_aura_option() );
+            stk::mesh::BulkData* meshBulk = meshBuilder.create( mSTKMeshData->mMtkMeshMetaData ).release();
 
             // Set member variable as pointer and bulk_data
             mSTKMeshData->mMtkMeshBulkData = ( meshBulk );
